### PR TITLE
Kmstest universalplanes etc

### PR DESCRIPTION
**Pull request by jsarha**
I almost forgot of these changes. The two review comments have been applied, also the patches "  kmstest: Reserve plane when we know the format of the FB we put in it" and  "kmstest: Separate plane reservation phase from command line parsing" have been squashed together. Keeping them separate would have meant more work and I did not see any point in that.

### CODE REVIEW

**Comment by tomba** [comment]
Travis builds are failing.

**Comment by jsarha** [comment]
On 12/19/17 17:12, Tomi Valkeinen wrote:
> Travis builds are failing.
> 
> 

/home/travis/build/tomba/kmsxx/utils/kmstest.cpp: In function ‘void
print_outputs(const std::vector<OutputInfo>&)’:

/home/travis/build/tomba/kmsxx/utils/kmstest.cpp:710:19: error: format
‘%llu’ expects argument of type ‘long long unsigned int’, but argument 3
has type ‘uint64_t {aka long unsigned int}’ [-Werror=format=]

           prop.val);

                   ^

Aaargh. It is impossible to printf uint64_t in a platform independent
way.... I need to come up with some ugly ifdef :(.



-- 
Texas Instruments Finland Oy, Porkkalankatu 22, 00180 Helsinki.
Y-tunnus/Business ID: 0615521-4. Kotipaikka/Domicile: Helsinki

**Comment by jsarha** [comment]
Found the fix and tested again... now I go to sleep :).

File: utils/kmstest.cpp [hunk 1]
```diff
@@ -631,9 +693,19 @@ static void print_outputs(const vector<OutputInfo>& outputs)
...
+		for (const PropInfo &prop: o.conn_props)
+			printf(" %s=%" PRIu64 ",", prop.prop->name().c_str(),
+			       prop.val);
+
+		printf("\n  Crtc %u/@%u", o.crtc->idx(), o.crtc->id());
+
+		for (const PropInfo &prop: o.crtc_props)
+			printf(" %s=%" PRIu64 ",", prop.prop->name().c_str(),
+			       prop.val);
+
```
Comment (tomba) [inline utils/kmstest.cpp]: For some reason this breaks the output:

```
# ./kmstest -c 1 -P DPMS=0 -r 640x480 -P zorder=2
Connector 1/@36: HDMI-A-1 DPMS=0,  Crtc 1/@40: 640x480 31.500 640/16/64/120 480/1/3/16 75 (75.00) 0xa 0x40
 zorder=2,  Plane 0/@31: 0,0-640x480
```

The trailing comma is also rather ugly there, and the format is different than with planes

File: utils/kmstest.cpp [hunk 2]
```diff
@@ -267,6 +281,26 @@ static void parse_plane(ResourceManager& resman, Card& card, const string& plane
...
+	if (split == string::npos)
+		EXIT("Equal sign ('=') not found in %s", prop_str.c_str());
+
+	name = prop_str.substr(0, split);
+	val = prop_str.substr(split+1);
+
+	prop = propobj->get_prop(name);
+	if (!prop)
+		EXIT("Property '%s' not found in obj id %u",
```
Comment (tomba) [inline utils/kmstest.cpp]: get_prop never returns null.

File: utils/kmstest.cpp [hunk 3]
```diff
@@ -311,10 +317,10 @@ static vector<Framebuffer*> get_default_fb(Card& card, unsigned width, unsigned
 	return v;
 }
 
-static vector<Framebuffer*> parse_fb(Card& card, const string& fb_str, unsigned def_w, unsigned def_h)
+static void parse_fb(ResourceManager& resman, Card& card, const string& fb_str, OutputInfo* output, PlaneInfo*& pinfo)
 {
```
Comment (tomba) [inline utils/kmstest.cpp]: Reference to pointer?

File: utils/kmstest.cpp [hunk 4]
```diff
@@ -311,10 +317,10 @@ static vector<Framebuffer*> get_default_fb(Card& card, unsigned width, unsigned
...
-	unsigned w = def_w;
-	unsigned h = def_h;
+	unsigned w = pinfo->w;
+	unsigned h = pinfo->h;
 	PixelFormat format = PixelFormat::XRGB8888;
```
Comment (tomba) [inline utils/kmstest.cpp]: Below you check for !pinfo, but here you always access pinfo.

File: utils/kmstest.cpp [hunk 5]
```diff
@@ -723,6 +730,7 @@ static void print_outputs(const vector<OutputInfo>& outputs)
 			for (const PropInfo &prop: p.props)
 				printf(" %s=%" PRIu64, prop.prop->name().c_str(),
 				       prop.val);
+			printf("\n");
```
Comment (tomba) [inline utils/kmstest.cpp]: I don't think this belongs to this patch

File: utils/kmstest.cpp [hunk 6]
```diff
@@ -700,21 +711,17 @@ static void print_outputs(const vector<OutputInfo>& outputs)
 			printf(" %s=%" PRIu64 ",", prop.prop->name().c_str(),
 			       prop.val);
 
-		printf("\n  Crtc %u/@%u", o.crtc->idx(), o.crtc->id());
+		printf("  Crtc %u/@%u", o.crtc->idx(), o.crtc->id());
```
Comment (tomba) [inline utils/kmstest.cpp]: Linefeed change here... Looks odd, why would this patch require linefeed changes here?

File: utils/kmstest.cpp (hunk 5 above)
Comment (jsarha) [reply utils/kmstest.cpp]: The universal planes and parsing order patch more or less affects a bit everything, also what makes sense in the printing phase... anyway these two patches affect each other quite a bit and keeping them separate is kind of hard. I already screwed up my first rebase fix attempt so badly, that I'll start from the scratch and try to make some sense to these.

**Comment by jsarha** [comment]
Now I hope this is finally ok.

**Comment by tomba** [comment]
Did you go through my comments? At least the "Reference to pointer" is still there.

**Comment by jsarha** [comment]
Applied trough tradition patch review process.

**Pull request closed without merging**